tests/tests/src/condition_compilation_test.res

```diff
@@ -15,11 +15,8 @@ type open_flag =
   | O_DSYNC
   | O_SYNC
   | O_RSYNC
-
   | O_SHARE_DELETE
-
   | O_CLOEXEC
-
   | O_KEEPEXEC
```

Comment:
With this PR, no newlines in between variant constructors. This makes the implementation simpler since I don't need to hack around node's location. 

Personally, I prefer the formatting this way too. What do you think? @cknitt @zth @nojaf

Reply:
Fine for me!

Reply:
This is great! 🎉

Reply:
Ok, I meant the new formatting of doc comments before variants is great.

Regarding the newlines, it would be nice if this was still possible:

```rescript
type x =
  // first group
  | A
  | B
  | C

  // second group
  | D
  | E
  | F
```

Reply:
@shulhi, this is better! I wasn't a fan of adjusting the ranges in the parser to satisfy the format. It felt wrong.